config/types.go

```diff
@@ -0,0 +1,22 @@
+package config
+
+// EndpointDiscoveryEnableState indicates if endpoint discovery is
+// enabled, disabled, or in default (auto) behavior state.
+//
+// Default behavior (AUTO) indicates operations that require endpoint
+// discovery will use Endpoint Discovery by default. Operations that
+// optionally use Endpoint Discovery will not use Endpoint Discovery
+// unless EndpointDiscovery is explicitly enabled.
+type EndpointDiscoveryEnableState uint
+
+// Enumeration values for EndpointDiscoveryEnableState
+const (
+	EndpointDiscoveryAuto     EndpointDiscoveryEnableState = iota // default state
+	EndpointDiscoveryDisabled                                     // endpoint discovery disabled
+	EndpointDiscoveryEnabled                                      // endpoint discovery enabled
+)
```

Comment:
Using these as string enums instead of int enums will simplify its overall usage and remove the need for the `ptr`. since the zero value of a string is `""` which is never a valid set value.

Reply:
```go
type EndpointDiscoveryEnableState string

// Enumeration values for EndpointDiscoveryEnableState
const (
	EndpointDiscoveryAuto      =  EndpointDiscoveryEnableState("auto") // default state
	EndpointDiscoveryDisabled  =  EndpointDiscoveryEnableState("false") // endpoint discovery disabled
	EndpointDiscoveryEnabled   =  EndpointDiscoveryEnableState("true")  // endpoint discovery enabled
)
```